utils/groovy/src/main/java/org/apache/brooklyn/util/groovy/GroovyJavaMethods.java

```diff
@@ -75,6 +76,20 @@
         }
     }
 
+    /**
+     * Alternative implementation of {@link ScriptBytecodeAdapter#isCase(Object, Object)}<br>
+     * Stripped down to work only for caseExpression of type <code>java.lang.Class</code>.<br>
```

Comment:
I wondered about that as well - that was my initial reaction. However, I'm fine with doing it like this for two reasons.

First, we can change most callers to instead use `JavaGroovyEquivalents` to avoid calling groovy code at all. Second, groovy allows some weird stuff with meta-programming, so I think it's safest to call the real Groovy code.